astropy/io/fits/column.py

```diff
@@ -1389,16 +1387,13 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
-            attr = []
-            for col in self:
-                val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+            attr = [getattr(col, cname) for col in self.columns]
+            if None in attr:
```

Comment:
Ok, looking more in detail, with a table with 720 columns, `__getattr__` is called 722 times for 'format' and 1441 times for 'name' ... and the result is always the same, so there something better to do.